### ISSUE TITLE
Receiving: ERROR_UNMATCH_CERTIFICATE_DECLARATION_IN_METADATA

### ISSUE THREAD

**Issue by 1hitsong** [tngan/samlify#394]
I'm using samlify version 2.7.5

I've looked at issue 269, but am unable to understand why I'm receiving this error.

My looking at the post SAML response back to the ACS controller, there is only 1 cert in a <ds:X509Certificate> tag, and only 1 cert in the IDP metadata. The cert data is the same for both.

Is there a way to debug this and see what cert values samlify is comparing? Perhaps if I could see the two cert strings it would help me debug this.

Any help would be appreciated.

**Comment by 1hitsong** [comment]
I ended up forking the samlify repo, updating the code to log the two cert strings, and using that repo instead of the npm installed one. 

I was then able to see the two certs and reverse engineer what was going wrong.

Closing issue.

**Comment by sibelius** [comment]
how did you solve the issue?

can you send a pull request?

**Comment by 1hitsong** [comment]
@sibelius Look at line 430 in https://github.com/tngan/samlify/blob/380d6126318c2460ce816557e6b5b49889f698eb/src/libsaml.ts

That's where the match check occurs. In my fork, I simply did a console.log of the two cert variables (cert, x509Certificate) so I could see what values were being compared. Once I had their values I was able to find the values back in the metadata to update the cert there to match.
